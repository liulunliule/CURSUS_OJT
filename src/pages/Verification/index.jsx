import React from 'react'
import "./index.scss";
import Verified from "../../assets/img/verification.png";
import dashboard from "../../assets/img/dashboard.png";
import { useSelector } from 'react-redux';

const Verification = () => {
    const isShowAll = useSelector((state) => state.savedCourse.isShowAll);
  return (
    <div className={`wrapper12 ${isShowAll ? "active" : ""}`}>
        <div className='sa4d25'>
            <div className='container-fluid'>
                <div className='row'>
                    <div className='col-lg-12'>
                        <h2 className='noti-title100'>
                            <img src={Verified} className=''/>
                            Verification
                        </h2>
                    </div>
                </div>

                <div className='row ver-sec'>
                    <div className='col-xl-6 col-lg-8 col-md-8'>
                        <div className='ver-cont'>
                            <img src='https://gambolthemes.net/html-items/cursus-new-demo/images/verified-account.svg'/>
                            <h4>Verification with Edututs+</h4>
                            <p>Praesent sed sapien gravida, tempus nunc nec, euismod turpis. Mauris quis scelerisque arcu. Quisque et aliquet nisl, id placerat est. Morbi quis imperdiet nulla.</p>
                            <ul className='ver-alert'>
                                <li>
                                    <div className='req_group'>
                                        <div className='dash_thingy'>
                                            <img src={dashboard}/>
                                        </div>
                                        <div className='edututs_required'>
                                            <span>
                                                <strong>14 subscribers</strong>
                                            </span>
                                            <span>500 required</span>

                                        </div>
                                    </div>
                                </li>

                                <li>
                                    <div className='req_group'>
                                        <div className='dash_thingy'>
                                            <img src={dashboard}/>
                                        </div>
                                        <div className='edututs_required'>
                                            <span>
                                                <strong>10 public watch hours</strong>
                                            </span>
                                            <span>1,500 required</span>
                                        </div>
                                    </div>
                                </li>

                            </ul>

                            <div className='verify_text'>
                                <img src={Verified} className=''/>
                                We'll send you an email when you're eligible to apply
                            </div>

                            <div className='form_ver'>
                                <h4>Verify Your ID</h4>
                                <form>
                                    <div className='random_title'>
                                        <label>Course Title*</label>
                                        <div class="input_tab_box">
											<input type="text" placeholder="Full Name" name="fullname" maxlength="60" id="full[name]" value="" />
										</div>
                                    </div>
                                    <div className='random_doc'>
                                        <label>Upload Document*</label>
                                        <div className='inputGroup'>
                                            <div className='customFile'>
                                                <input type="file" class="custom-file-input" id="inputGroupFile06" />
                                                <label class="custom-file-label" for="inputGroupFile06">No Choose</label>
                                            </div>
                                        </div>
                                    </div>
                                    <button class="verify_submit_btn" type="submit">Submit Now</button>
                                </form>

                            </div>
                        </div>

                    </div>

                </div>
            </div>
        </div>
    </div>
  )
}

export default Verification