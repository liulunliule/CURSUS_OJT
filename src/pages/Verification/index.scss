.noti-title100 {
    font-size: 20px;
    margin-top: 0 !important;
    margin-bottom: 8px !important;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #333;
    margin-left: 20px;

    img {
        width: 25px;
        height: 25px;
        margin-right: 10px;
    }
}
.ver-sec {
    justify-content: center;
}

.ver-cont {
    text-align: center;
    margin-top: 30px;

    img {
        width: 200px;
    }

    h4 {
        font-size: 24px;
        font-weight: 600;
        font-family: 'Roboto', sans-serif;
        text-align: center;
        color: #333;
        margin-top: 40px !important;
        margin-bottom: 24px !important;
    }

    p {
        font-size: 14px;
        font-weight: 400;
        font-family: 'Roboto', sans-serif;
        text-align: center;
        color: #686f7a;
        line-height: 26px;
        margin-bottom: 0;
    }
}

.ver-alert {
    background: #fff;
    border: 1px solid #efefef;
    border-radius: 3px;
    margin-top: 30px;

    li {
        display: inline-block;
        width: 50%;
        margin-left: -2px;
        border-right: 1px solid #efefef;
        border-bottom: 0;
    }
}

.req_group {
    display: flex;
    padding: 20px;
    align-items: center;
}

.dash_thingy img {
    width: 30px;
    height: 30px;
}

.edututs_required span {
    display: block;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    margin-left: 10px;
    margin-bottom: 8px;
    text-align: left;
}

.verify_text {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    color: #686f7a;
    line-height: 26px;
    margin-top: 20px;
    margin-bottom: 0;

    img {
        width: 25px;
        height: 25px;
        // margin-right: 10px;
    }
}

.form_ver {
    background: #fff;
    border: 1px solid #efefef;
    border-radius: 3px;
    padding: 0 30px 30px;
    margin-top: 50px;
    margin-bottom: 30px;

    h4 {
        font-size: 24px;
        font-weight: 600;
        font-family: 'Roboto', sans-serif;
        text-align: center;
        color: #333;
        margin-top: 40px !important;
        margin-bottom: 24px !important;
    }
}

.random_title {
    margin-top: 50px;
    position: relative;

    label {
        font-size: 14px;
        font-weight: 500;
        font-family: 'Roboto', sans-serif;
        margin-bottom: 10px !important;
        color: #333;
        text-align: left;
        display: block;
    }
}

.input_tab_box {
    position: relative;
    font-weight: 400;
    font-style: normal;
    display: inline-flex;
    color: rgba(0,0,0,.87);
    font-size: 14px !important;
    width: 100%;

    input {
        padding-left: 1em!important;
        padding-right: 2em!important;
        height: 40px !important;
        font-size: 14px;
        font-weight: 400;
        font-family: 'Roboto', sans-serif;
        border-radius: 3px !important;
        flex: 1 0 auto;
        border: 1px solid rgba(34, 36, 38, .15);

    }
}

.random_doc {
    margin-top: 30px;

    label {
        font-size: 14px;
        font-weight: 500;
        font-family: 'Roboto', sans-serif;
        margin-bottom: 10px !important;
        color: #333;
        text-align: left;
        display: block;
    }
}

.inputGroup {
    position: relative;
    display: flex;
    flex-wrap: wrap;
    align-items: stretch;
    width: 100%;
}

.customFile {
    position: relative;
    -ms-flex: 1 1 auto;
    flex: 1 1 auto;
    width: 1%;
    min-width: 0;
    margin-bottom: 0;
    background: #f7f7f7;

}

.custom-file-label {
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    z-index: 1;
    height: calc(1.5em + .75rem + 2px);
    padding: 1.3rem 2rem;
    font-weight: 100;
    line-height: 0;
    color: #686f7a;
    background-color: #fff;
    border: 1px solid #efefef;
    border-radius: 3px;
    margin-bottom: 0;
    box-shadow: inset 0px 1px 4px rgba(0, 0, 0, 0.02) !important;
}

.verify_submit_btn {
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #fff;
    background: #ed2a26;
    padding: 0px 30px;
    border-radius: 3px;
    border: 0;
    height: 40px;
    margin-top: 45px;
    width: 100%;

    &:hover {
        background: black;
    }
}