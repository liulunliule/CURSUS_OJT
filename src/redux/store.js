import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
