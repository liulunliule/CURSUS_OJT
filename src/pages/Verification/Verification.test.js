import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Verification from '.';

test('test verification page', () => {
  const tree = renderer.create(
    <Provider store={store}>
        <MemoryRouter>
            <Verification/>
        </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});