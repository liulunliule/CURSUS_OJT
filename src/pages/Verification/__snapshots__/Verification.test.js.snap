// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Certification Test Result Page 1`] = `
<div
  className="wrapper12 "
>
  <div
    className="sa4d25_sa"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <h2
            className="noti-title100"
          >
            <img
              className=""
              src="verification.png"
            />
            Verification
          </h2>
        </div>
      </div>
      <div
        className="row ver-sec"
      >
        <div
          className="col-xl-6 col-lg-8 col-md-8"
        >
          <div
            className="ver-cont"
          >
            <img
              src="https://gambolthemes.net/html-items/cursus-new-demo/images/verified-account.svg"
            />
            <h4>
              Verification with Edututs+
            </h4>
            <p>
              Praesent sed sapien gravida, tempus nunc nec, euismod turpis. Mauris quis scelerisque arcu. Quisque et aliquet nisl, id placerat est. Morbi quis imperdiet nulla.
            </p>
            <ul
              className="ver-alert"
            >
              <li>
                <div
                  className="req_group"
                >
                  <div
                    className="dash_thingy"
                  >
                    <img
                      src="dashboard.png"
                    />
                  </div>
                  <div
                    className="edututs_required"
                  >
                    <span>
                      <strong>
                        14 subscribers
                      </strong>
                    </span>
                    <span>
                      500 required
                    </span>
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="req_group"
                >
                  <div
                    className="dash_thingy"
                  >
                    <img
                      src="dashboard.png"
                    />
                  </div>
                  <div
                    className="edututs_required"
                  >
                    <span>
                      <strong>
                        10 public watch hours
                      </strong>
                    </span>
                    <span>
                      1,500 required
                    </span>
                  </div>
                </div>
              </li>
            </ul>
            <div
              className="verify_text"
            >
              <img
                className=""
                src="verification.png"
              />
              We'll send you an email when you're eligible to apply
            </div>
            <div
              className="form_ver"
            >
              <h4>
                Verify Your ID
              </h4>
              <form>
                <div
                  className="random_title"
                >
                  <label>
                    Course Title*
                  </label>
                  <div
                    class="input_tab_box"
                  >
                    <input
                      id="full[name]"
                      maxlength="60"
                      name="fullname"
                      placeholder="Full Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="random_doc"
                >
                  <label>
                    Upload Document*
                  </label>
                  <div
                    className="inputGroup"
                  >
                    <div
                      className="customFile"
                    >
                      <input
                        class="custom-file-input"
                        id="inputGroupFile06"
                        type="file"
                      />
                      <label
                        class="custom-file-label"
                        for="inputGroupFile06"
                      >
                        No Choose
                      </label>
                    </div>
                  </div>
                </div>
                <button
                  class="verify_submit_btn"
                  type="submit"
                >
                  Submit Now
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test verification page 1`] = `
<div
  className="wrapper12 "
>
  <div
    className="sa4d25_sa"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <h2
            className="noti-title100"
          >
            <img
              className=""
              src="verification.png"
            />
            Verification
          </h2>
        </div>
      </div>
      <div
        className="row ver-sec"
      >
        <div
          className="col-xl-6 col-lg-8 col-md-8"
        >
          <div
            className="ver-cont"
          >
            <img
              src="https://gambolthemes.net/html-items/cursus-new-demo/images/verified-account.svg"
            />
            <h4>
              Verification with Edututs+
            </h4>
            <p>
              Praesent sed sapien gravida, tempus nunc nec, euismod turpis. Mauris quis scelerisque arcu. Quisque et aliquet nisl, id placerat est. Morbi quis imperdiet nulla.
            </p>
            <ul
              className="ver-alert"
            >
              <li>
                <div
                  className="req_group"
                >
                  <div
                    className="dash_thingy"
                  >
                    <img
                      src="dashboard.png"
                    />
                  </div>
                  <div
                    className="edututs_required"
                  >
                    <span>
                      <strong>
                        14 subscribers
                      </strong>
                    </span>
                    <span>
                      500 required
                    </span>
                  </div>
                </div>
              </li>
              <li>
                <div
                  className="req_group"
                >
                  <div
                    className="dash_thingy"
                  >
                    <img
                      src="dashboard.png"
                    />
                  </div>
                  <div
                    className="edututs_required"
                  >
                    <span>
                      <strong>
                        10 public watch hours
                      </strong>
                    </span>
                    <span>
                      1,500 required
                    </span>
                  </div>
                </div>
              </li>
            </ul>
            <div
              className="verify_text"
            >
              <img
                className=""
                src="verification.png"
              />
              We'll send you an email when you're eligible to apply
            </div>
            <div
              className="form_ver"
            >
              <h4>
                Verify Your ID
              </h4>
              <form>
                <div
                  className="random_title"
                >
                  <label>
                    Course Title*
                  </label>
                  <div
                    class="input_tab_box"
                  >
                    <input
                      id="full[name]"
                      maxlength="60"
                      name="fullname"
                      placeholder="Full Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="random_doc"
                >
                  <label>
                    Upload Document*
                  </label>
                  <div
                    className="inputGroup"
                  >
                    <div
                      className="customFile"
                    >
                      <input
                        class="custom-file-input"
                        id="inputGroupFile06"
                        type="file"
                      />
                      <label
                        class="custom-file-label"
                        for="inputGroupFile06"
                      >
                        No Choose
                      </label>
                    </div>
                  </div>
                </div>
                <button
                  class="verify_submit_btn"
                  type="submit"
                >
                  Submit Now
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
